doc/k8s/k8s.md

```diff
@@ -0,0 +1,199 @@
+k8s integration design document
+---
+
+## Overview
+
+[Kubernetes](https://kubernetes.io/) (k8s) is an open source container orchestrator
+which automates deployment, scaling and management of containerized applications.
+By letting k8s manages the containers running in SONiC, we can manage the fleet of
+SONiC switches in a centralized way.
+
+k8s has many features to ease the operation of daily containerized applications.
+One example is rolling update. By using k8s, the operator can upgrade all container
+running in SONiC image with a simple instruction. Stopping the current container and
+starting the new updated container can be done automatically in parallel in the fleet of SONiC switches.
+
+## Glossary
+
+Based on [k8s glossary](https://kubernetes.io/docs/reference/glossary/?fundamental=true)
+
+- Cluster
+    - A set of machines, called nodes, that run containerized applications managed by Kubernetes
+    - In SONiC use-case, each machine is SONiC switch
```

Comment:
This statement looks weird. Do you mean cluster?

Reply:
A cluster is a set of machines and each machine is a SONiC switch (except controller node).
How should I change the statement?

Reply:
I guess you mean 'In SONiC use-case, a node is a SONiC switch. And a cluster is all the SONiC switches managed by Kubernetes'.